src/cli/README.md

```diff
@@ -758,6 +758,30 @@ Send DNS Query to obtain IPv6 address for given hostname. The latter two paramet
 > DNS response for ipv6.google.com - 2a00:1450:401b:801:0:0:0:200e TTL: 300
 ```
 
+### dns browse \<service-name\> \[DNS server IP\] \[DNS server port\]
+
+Send a browse (service instance enumeration) DNS query to get the list of services for given service-name.
+
+```bash
+> dns browse _service._udp.example.com
+DNS browse response for _service._udp.example.com.
+inst1
+    Port:1234, Priority:1, Weight:2, TTL:7200
+    Host:host.example.com.
+    HostAddress:fd00:0:0:0:0:0:0:abcd TTL:7200
+    TXT-Data:(len:11) [096b65793d76616c756500] TTL:7300
```

Comment:
We added methods to parse the TXT records into `otDnsTxtEntry` in https://github.com/openthread/openthread/pull/6113, it would be better to have parsed TXT entries in the CLI.

Reply:
I actually wanted to use it, but noticed that the main public API is `otSrpServerServiceGetNextTxtEntry()` and is under `SRP_SERVER` config. The DNS client may be enabled/used without SRP server feature, so we cannot use this API.

I think later (in a new PR) we should move the TXT record parsing API under common `dns.h` and the implementation in `dns_header.cpp` so we can use it from all modules (independent of which features are enabled/disabled).

Reply:
`otSrpServerServiceGetNextTxtEntry` is a simple wrapper of `Dns::TxtRecord::GetNextTxtEntry`. I think we can also have an DNS client API like `otDnsBrowseResponseGetNextTxtNetry` which is also simply a wrapper of `Dns::TxtRecord::GetNextTxtEntry`. Thoughts?

Reply:
A while after I added the comment, I actually also thought of the same model as you suggested 😃 (i.e., if we can add a `otDnsBrowseResponseGetNextTxtEntry` for `BrowseResponse` itself using `TxtRecord::GetNextEntry()`)

I don't think that it will be straight-forward:
- The `GetNextEntry()` requires `aTxtData` array to be given to it so it can parse through the entries. 
- In the case of `otDnsBrowseResponse`, we won't have the txt data in an array but we will have it in the response `Message`. 
- I don't see any easy way to extend `GetNextEntry()` to work with `Message` directly (since we use pointers to `aTxtData` array to get the "key" and "value" segment - which is a good model in my opinion).

The way I was/am considering is that we add a new common `ot` public function (in `dns.h`) like `otDnsGetNextTxtEntry(otDnsTxtEntry *aIterator, const char *aTxtData, uint16_t aTxtDataLength)` to parse any given txt data. This then makes it the caller's job to provide the txt data buffer and its content. The other upside I see is that the same `ot` API can be used for different cases (e.g., browse response or service response) and even maybe for SRP server services.

Thoughts? Other ideas?

Reply:
oh, yes we are retrieving data directly from the response message, agree that there is no easy way to have `otDnsBrowseResponseGetNextTxtEntry` and a public API `otDnsGetNextTxtEntry` should be what we need :)

Reply:
BTW, @wgtdkp just letting you know so we don't both do the same thing. 😄 

I already started adding this (i.e. a common txt record parse api) - will also add a bunch of unit tests for the whole txt entry encoding/decoding in the same PR.

Reply:
@abtink Thanks! :+1: 
Looking forward to your PR!

Reply:
We have this now in PR #6157.